tests/Message/MessageBagTest.php

```diff
@@ -125,6 +128,10 @@ public function jsonSerialize(): void
                 ['role' => 'system', 'content' => 'My amazing system prompt.'],
                 ['role' => 'assistant', 'content' => 'It is time to sleep.'],
                 ['role' => 'user', 'content' => 'Hello, world!'],
+                ['role' => 'user', 'content' => 'Analyze the following image:'],
+                ['role' => 'user', 'content' => [
+                    ['type' => 'image_url', 'image_url' => ['url' => 'http://example.com/image.jpg']],
+                ]],
```

Comment:
The only difference I propose is to use 2 different messages, would that be feasible for you @DZunke?

Reply:
with this we won't get the structure that is conform to the api specs:

```json
{
    "model": "gpt-4o-mini",
    "messages": [
      {
        "role": "user",
        "content": [
          {
            "type": "text",
            "text": "What’s in this image?"
          },
          {
            "type": "image_url",
            "image_url": {
              "url": "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/2560px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg"
            }
          }
        ]
      }
    ],
    "max_tokens": 300
  }
```
see https://platform.openai.com/docs/guides/vision

Reply:
same for claude:

```json
{"role": "user", "content": [
  {
    "type": "image",
    "source": {
      "type": "base64",
      "media_type": "image/jpeg",
      "data": "/9j/4AAQSkZJRg...",
    }
  },
  {"type": "text", "text": "What is in this image?"}
]}
```
see https://docs.anthropic.com/en/api/messages

Reply:
Thanks for pointing it out with another design but yeah. As Christopher mentioned it would not fit the specification and also change the functionality to split it up because the text is the prompt how the image should be tackled as far as i understood it. A previous message would just be "context" and irrelevant in itself for the image processing.

Reply:
It would not fulfill the spec, but yes it generates a valid answer as you can see in my screenshot above.